tests/test_core_db_coverage.py

```
@@ -171,3 +176,61 @@ def test_sqlite_file_path_creation(self):
+        result = _derive_async_url("sqlite:///test.db")
+        assert result == "sqlite+aiosqlite:///test.db"
+
+    def test_async_engine_not_configured(self):
+        """Test that async_engine is None when not configured."""
+        assert async_engine is None
+
+    @pytest.mark.asyncio
+    async def test_get_async_session_not_configured(self):
+        """Test get_async_session raises error when not configured."""
+        with pytest.raises(RuntimeError, match="Async SQLAlchemy is not configured"):
+            async for session in get_async_session():
```

Ruff (0.13.3)

Loop control variable `session` not used within loop body

Rename unused `session` to `_session`

(B007)